src/train_tracker.py

```diff
@@ -0,0 +1,228 @@
+"""列車捕捉モジュール.
+
+列車を捕捉する
+@author: desty505 miyashita64
+"""
+
+import cv2
+import numpy as np
+from camera_interface import CameraInterface
+
+
+class TrainTracker:
+    """列車を捕捉するクラス."""
+
+    def __init__(self, camera_id: int) -> None:
+        """コンストラクタ.
+
+        Args:
+            camera_id (int): カメラID
+        """
+        self.camera = CameraInterface(camera_id)
+        self.diff_border = 20
+        self.observe_rect_points = [(0, 0), (0, 0)]
+
+    def __del__(self) -> None:
+        """デストラクタ."""
+        del self.camera
+
+    def calibrate(self) -> None:
+        """キャリブレーション."""
+        # 録画を開始する
+        video, mark_video = self.camera.start_record()
+        # クリックに対するコールバックを設定
+        cv2.namedWindow("Calibration")
+        cv2.setMouseCallback("Calibration", self.mouse_callback)
+
+        while True:
+            # フレームを取得
+            success, frame = self.camera.get_frame()
+            if not success:
+                break
+
+            # クリックされた領域の描画
+            mark_frame = self.draw_observe_rect(frame)
+
+            # フレームの表示
+            cv2.imshow("Calibration", mark_frame)
+            # フレームのファイル出力
+            video.write(frame)              # 撮影した動画
+            mark_video.write(mark_frame)    # 輪郭線を描画した動画
+
+            # "q"キーを押すとループを抜けて終了
+            if cv2.waitKey(1) & 0xFF == ord("q"):
+                break
+
+        # リソースを解放
+        self.camera.end_record()
+
+    def mouse_callback(self, event, x, y, flags, param) -> None:
+        """キャリブレーションのクリック時の処理.
+
+        Args:
+            event: クリックイベント
+            x: クリックのx座標
+            y: クリックのy座標
+            flags: 追加情報のフラグ
+            param: 任意パラメータ
+        """
+        if event == cv2.EVENT_LBUTTONDOWN:
+            # クリック時の座標を記録
+            self.observe_rect_points.append((x, y))
+
+    def draw_observe_rect(self, frame) -> np.ndarray:
+        """監視する領域をフレームに描画する.
+
+        Args:
+            frame: 描画対象フレーム
+        Returns:
+            mark_frame: 監視する領域を描画したフレーム
+        """
+        mark_frame = frame.copy()
+        cv2.rectangle(mark_frame, self.observe_rect_points[-1],
```

Comment:
```suggestion
        # 最後に指定した2か所の座標から長方形を描画する
        cv2.rectangle(mark_frame, self.observe_rect_points[-1],
```
というか、mouse_callback()で古い座標は消していった方がきれい？